cirq-core/cirq/circuits/circuit_operation.py

```diff
@@ -294,25 +346,29 @@ def _parameter_names_generator(self) -> Iterator[str]:
             ):
                 yield name
 
+    @cached_property
+    def _mapped_any_loop(self) -> 'cirq.Circuit':
+        circuit = self.circuit.unfreeze()
+        if self.qubit_map:
+            circuit = circuit.transform_qubits(lambda q: self.qubit_map.get(q, q))
+        if isinstance(self.repetitions, INT_CLASSES) and self.repetitions < 0:
+            circuit = circuit**-1
+        if self.measurement_key_map:
+            circuit = protocols.with_measurement_key_mapping(
+                circuit, dict(self.measurement_key_map)
+            )
+        if self.param_resolver:
+            circuit = protocols.resolve_parameters(
+                circuit, self.param_resolver, recursive=False
+            )
+        return cast(circuits.Circuit, circuit)
```

Comment:
Do we have to return `cirq.Circuit` here or could this be `cirq.AbstractCircuit`? If the former then I'd suggest unfreezing instead:
```suggestion
        return circuit.unfreeze(copy=False)
```